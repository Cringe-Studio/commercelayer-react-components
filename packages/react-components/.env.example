# Unit tests
VITE_TEST_CLIENT_ID=
VITE_TEST_ENDPOINT=
VITE_TEST_MARKET_ID=
VITE_TEST_USERNAME=
VITE_TEST_PASSWORD=
VITE_TEST_CLIENT_ID_INTEGRATION=
VITE_TEST_CLIENT_SECRET=
