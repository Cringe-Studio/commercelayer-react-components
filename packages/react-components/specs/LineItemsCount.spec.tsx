import React from 'react'
import { LineItemsCount } from '../packages/react-components/src'
import renderer from 'react-test-renderer'
import components from '../#config/components'
import Parent from '../#components/utils/Parent'

const propTypes = components.LineItemsCount.propTypes

test('<LineItemsCount/>', () => {
  expect.assertions(2)
  const component = renderer.create(<LineItemsCount />)
  const tree = component.toJSON()
  const root = component.toTree()
  const proptypes = root.type['propTypes']
  expect(tree).toMatchSnapshot()
  expect(proptypes.children).toBe(propTypes.children)
})

test('<LineItemsCount check children format />', () => {
  expect.assertions(2)
  const component = renderer.create(<LineItemsCount />)
  const tree = component.toJSON()
  const root = component.toTree()
  const rendered = root.rendered
  expect(tree).toMatchSnapshot()
  expect(rendered.type).toBe('span')
})

test('<LineItemsCount with custom children />', () => {
  expect.assertions(6)
  const CustomComponent = (props) => <span>{props.label}</span>
  const component = renderer.create(
    <LineItemsCount>{CustomComponent}</LineItemsCount>
  )
  const tree = component.toJSON()
  const root = component.toTree()
  const rendered = root.rendered
  const childRendered = root.rendered.rendered
  expect(tree).toMatchSnapshot()
  expect(rendered.type).toBe(Parent)
  expect(childRendered.nodeType).toBe('component')
  expect(childRendered.type).toBe(CustomComponent)
  expect(childRendered.rendered.nodeType).toBe('host')
  expect(childRendered.rendered.type).toBe('span')
})
