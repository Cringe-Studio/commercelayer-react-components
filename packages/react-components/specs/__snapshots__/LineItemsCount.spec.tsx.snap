// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LineItemsCount check children format /> 1`] = `
<span>
  0
</span>
`;

exports[`<LineItemsCount with custom children /> 1`] = `<span />`;

exports[`<LineItemsCount/> 1`] = `
<span>
  0
</span>
`;
